plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStoreMeta.php

```diff
@@ -0,0 +1,180 @@
+	 *
+	 * @return array Array elements: table, object_id_field, meta_id_field.
+	 */
+	protected function get_db_info() {
+		global $wpdb;
+
+		return array(
+			'table'           => OrdersTableDataStore::get_meta_table_name(),
+			'meta_id_field'   => 'id',
+			'object_id_field' => 'order_id',
+		);
+	}
+
+	/**
+	 * Returns an array of meta for an object.
+	 *
+	 * @param  WC_Data $object WC_Data object.
+	 * @return array
+	 */
+	public function read_meta( &$object ) {
+		global $wpdb;
+
+		$db_info = $this->get_db_info();
+
+		// XXX: do we want to augment this with pre-COT metadata?
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$raw_meta_data = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT {$db_info['meta_id_field']} AS meta_id, meta_key, meta_value FROM {$db_info['table']} WHERE {$db_info['object_id_field']} = %d ORDER BY meta_id",
+				$object->get_id()
+			)
+		);
+		// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+
+		return $raw_meta_data;
+	}
+
+	/**
+	 * Deletes meta based on meta ID.
+	 *
+	 * @param  WC_Data  $object WC_Data object.
+	 * @param  stdClass $meta (containing at least ->id).
+	 */
+	public function delete_meta( &$object, $meta ) {
+		global $wpdb;
+
+		if ( ! is_a( $meta, 'WC_Meta_Data' ) || ! isset( $meta->id ) ) {
+			return false;
+		}
+
+		$db_info = $this->get_db_info();
+
+		$meta_id      = absint( $meta->id );
+		$current_meta = $this->get_metadata_by_id( $meta_id );
+		if ( $current_meta ) {
+			return (bool) $wpdb->delete( $db_info['table'], array( $db_info['meta_id_field'] => $meta_id ) );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Add new piece of meta.
+	 *
+	 * @param  WC_Data  $object WC_Data object.
+	 * @param  stdClass $meta (containing ->key and ->value).
+	 * @return int meta ID
+	 */
+	public function add_meta( &$object, $meta ) {
+		global $wpdb;
+
+		if ( ! is_a( $meta, 'WC_Meta_Data' ) ) {
+			return false;
+		}
+
+		$db_info = $this->get_db_info();
+
+		$object_id  = $object->get_id();
+		$meta_key   = wp_unslash( wp_slash( $meta->key ) );
+		$meta_value = maybe_serialize( is_string( $meta->value ) ? wp_unslash( wp_slash( $meta->value ) ) : $meta->value );
+
+		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_value,WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+		$result = $wpdb->insert(
+			$db_info['table'],
+			array(
+				$db_info['object_id_field'] => $object_id,
+				'meta_key'                  => $meta_key,
+				'meta_value'                => $meta_value,
+			)
+		);
+		// phpcs:enable WordPress.DB.SlowDBQuery.slow_db_query_meta_value,WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+
+		return $result ? (int) $wpdb->insert_id : false;
+	}
+
+	/**
+	 * Update meta.
+	 *
+	 * @param  WC_Data  $object WC_Data object.
+	 * @param  stdClass $meta (containing ->id, ->key and ->value).
+	 */
+	public function update_meta( &$object, $meta ) {
+		global $wpdb;
+
+		$current_meta = $this->get_metadata_by_id( $meta->id );
```

Comment:
same as above, we don't need to fetch, but directly run the query and return output based on results.

Reply:
Done.